reporter/config.go

```diff
@@ -51,4 +52,25 @@ type Config struct {
 
 	// gRPCInterceptor is the client gRPC interceptor, e.g., for sending gRPC metadata.
 	GRPCClientInterceptor grpc.UnaryClientInterceptor
+
+	// ExtraSampleAttrProd is an optional hook point for adding custom
+	// attributes to samples.
+	ExtraSampleAttrProd SampleAttrProducer
+}
+
+// SampleAttrProducer provides a hook point to:
+//
+// - inspect each trace and its meta when it is enqueued in the reporter
+// - produce extra meta info
+// - attach extra attributes to the trace
+type SampleAttrProducer interface {
```

Comment:
Should this be moved to `iface.go`?

Reply:
I considered that originally, but `iface.go` is currently dedicated to the public interface implemented by any reporter. If found `config.go` to be a better fit. I didn't want readers to incorrectly conclude that they have to implement this for a reporter.